### DISCUSSION TITLE
MongoDB driver

### DISCUSSION THREAD

**Discussion by ermya** [swoole/swoole-src#5164] [Q&A] [answered]
Hello
Is there a way to communicate asynchronously with MongoDB?
Is it possible to hook the MongoDB PHP driver and use it asynchronously?
Is there a connection pool for MongoDB?

**Comment by twose** [comment, marked as answer]
The MongoDB official favors multi-threading mode, so there is no support for multiplexing, which is one reason.
The MySQL clients in PHP such as PDO_mysql, mysqli, etc. are based on the mysqlnd driver developed by PHP itself. mysqlnd is based on php_stream, which provides a good abstraction for socket operations, hence it can be easily hooked.

I’d highly encourage to cast your votes on the MongoDB Feedback Forum for this request - [Asynchronous variant of MongoDB C Driver – MongoDB Feedback Engine](https://feedback.mongodb.com/forums/924286-drivers/suggestions/47080057-asynchronous-variant-of-mongodb-c-driver) .

https://www.mongodb.com/community/forums/t/why-not-supply-async-api-in-mongo-c-driver/16260

**Reply by rbro** [reply]
I recently found this:

https://www.mongodb.com/docs/languages/c/c-driver/current/async-c-driver/

Once this is out of preview, would this allow calls to MongoDB from swoole?
